esb/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/part/EsbMultiPageEditor.java

```diff
@@ -667,16 +698,20 @@ private void updateAssociatedDiagrams() {
     public void doSave(IProgressMonitor monitor) {
     	//Fixing TOOLS-2958
     	setContextClassLoader();
-    	if(getActivePage()==SOURCE_VIEW_PAGE_INDEX){
-    		handleDesignViewActivatedEvent();
-    	}
-    	sourceDirty=false;
-        getEditor(0).doSave(monitor);
-        EsbServer esbServer = EditorUtils.getEsbServer(graphicalEditor);
-        
-        //Since Complex endpoint type editors dose not have assiociated xml file do not need to call this.
-        //if(!esbServer.getType().equals(ArtifactType.COMPLEX_ENDPOINT)){
-
+		sourceDirty = false;
+		if (getActivePage() == SOURCE_VIEW_PAGE_INDEX) {
+			try {
+				handleDesignViewActivatedEvent();
+			} catch (Exception e) {
```

Comment:
handleDesignViewActivatedEvent() method throws Exception and it is inherited from internal methods.